apps/base-docs/tutorials/docs/0_gassless-transactions-with-paymaster.md

```diff
@@ -0,0 +1,616 @@
+
+To do so, we'll first need two private keys that will be used to create the Smart Accounts using Foundry. Open a terminal and run the following command **twice**:
+
+```bash
+cast wallet new
+```
+
+You should see something like this:
+
+```bash
+> cast wallet new
+Successfully created new keypair.
+Address: 0xD440D74620542...D6F005cfD9
+Private key: 0x01c9720c1dfa3c9...634793138897
+
+
+> cast wallet new
+Successfully created new keypair.
+Address: 0x5f8e5bC8620542...D6F005cfD9
+Private key: 0xbcd6fbc1dfa3c9...634793138897
+```
+
+Use the newly generated private keys as the parameter to the `privateKeyToSimpleSmartAccount` function and set the factory address to
+
+```javascript
+const simpleAccount = await privateKeyToSimpleSmartAccount(publicClient, {
+  privateKey: '0xD440D74620542bb7B0c70922136e6d25ef2030d7ecd7c9061a463479313889786',
+  factoryAddress: baseFactoryAddress, //Base v0.6
+  entryPoint: baseEntryPoint,
+});
+
+const simpleAccount2 = await privateKeyToSimpleSmartAccount(publicClient, {
+  privateKey: '0x5f8e5bC86205421d970922136e6asda5ef2030d7ecd7c9061a46347931385701b',
+  factoryAddress: baseFactoryAddress, //Base v0.6
+  entryPoint: baseEntryPoint,
+});
+```
+
+Let’s break down what’s happening here:
+Private key : the private key to the wallet that your created either using Foundry or a wallet that you own. Either is fine.
+
+Factory address is the address to the smart account factory deployed on base. More details here.
+Entrypoint is the entrypoint contract for base. More details here.
+
+### Initialize Paymaster and Create account
+
+Initialize the paymaster and smart account client for both smart accounts:
+
+```javascript
+const cloudPaymaster = createPimlicoPaymasterClient({
+  chain: base,
+  transport: http(rpcUrl),
+  entryPoint: baseEntryPoint,
+});
+
+const smartAccountClient = createSmartAccountClient({
+  account: simpleAccount,
+  chain: base,
+  bundlerTransport: http(rpcUrl),
+  middleware: {
+    sponsorUserOperation: cloudPaymaster.sponsorUserOperation,
+  },
+});
+
+const smartAccountClient2 = createSmartAccountClient({
+  account: simpleAccount2,
+  chain: base,
+  bundlerTransport: http(rpcUrl),
+  middleware: {
+    sponsorUserOperation: cloudPaymaster.sponsorUserOperation,
+  },
+});
+```
+
+To sponsor transactions we will need the ABI and the address of the contracts we want to support.
+
+:::note Sponsor any contract
+Feel free to use your own contract to interact with the Paymaster. For learning purposes, we will interact with an ERC-721 contract deployed on the Base mainnet. We will call the mintTo function, which takes an address as its only parameter.
+:::
+
+We will be interacting with the NFT + ABI from a simple NFT contract deployed at: `0x83bd615eb93eE1336acA53e185b03B54fF4A17e8`
+
+**Copy and paste the NFT's abi into `index.js`**
```

Comment:
I'd put this in a separate file and import it

Reply:
Good suggestion and I would do this in production but for the sake of the tutorial, i'm going to leave it bc i'd still have to paste the ABI and tell them to create a new file and import it.

I could tell them to copy it from basescan but would rather them not move around too much.